### PR TITLE
Standalone mode

### PR DESCRIPTION
**Pull request by anna1901**

### CODE REVIEW

File: AirCasting/ABConnector/BluetoothManager.swift [hunk 1]
```diff
@@ -102,6 +102,7 @@ extension BluetoothManager: CBCentralManagerDelegate {
     func centralManager(_ central: CBCentralManager, didDiscover peripheral: CBPeripheral, advertisementData: [String : Any], rssi RSSI: NSNumber) {
         if !devices.contains(peripheral) {
             if peripheral.name != nil {
+                // check if ab is not recording stand alone session at the moment
                 devices.append(peripheral)
             }
         }
```
Comment (jankrzempek) [inline AirCasting/ABConnector/BluetoothManager.swift:108]: I know it's not yours but could you please change it to a one-liner if statement, please?

File: AirCasting/SessionViews/SessionCardView.swift [hunk 2]
```diff
@@ -62,7 +62,11 @@ struct SessionCartView: View {
     }
     
     var body: some View {
-        sessionCard
+        if session.isInStandaloneMode {
+            standaloneSessionCard
+        } else {
+            sessionCard
+        }
```
Comment (jankrzempek) [inline AirCasting/SessionViews/SessionCardView.swift:69]: Maybe one-liner `session.isInStandaloneMode ? standaloneSessionCard : sessionCard

File: AirCasting/SessionViews/StandaloneSessionCardView.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 5) {
+            header
+            VStack(spacing: 15) {
+                Text("Your AirBeam3 is now in stand-alone mode")
+                    .font(Fonts.boldHeading3)
+                    .foregroundColor(.darkBlue)
+                Text("AirBeam3 is now recording using its SD card. The measurements will be displayed here after syncing.")
+                Button("Finish recording & don't sync") {
```
Comment (jankrzempek) [inline AirCasting/SessionViews/StandaloneSessionCardView.swift:20]: Those text to Strings struct please 😄

**Review by jankrzempek — CHANGES_REQUESTED** [review]
Those Strings please and others are only suggestions. Cool code 🚀

File: AirCasting/SessionViews/SessionCardView.swift (hunk 2 above)
Comment (pawelgil) [reply AirCasting/SessionViews/SessionCardView.swift:69]: I think you can't do that inside `ViewBuilders`. I still do those in one line tho:
```Swift
if session.isInStandaloneMode { standaloneSessionCard } else { sessionCard }
```

**Review by pawelgil — CHANGES_REQUESTED** [review]
Some cosmetic changes needed, otherwise nice job! (But this whole `BluetoothManager`, `PeripheralSessionManager`, `AirBeamConfigurator` etc. gang is getting out of hand, we'll need to refactor this really soon in order not to drown in spaghetti)

File: AirCasting/SessionViews/SessionHeaderView.swift [hunk 4]
```diff
@@ -135,6 +140,14 @@ private extension SessionHeaderView {
         }
     }
     
+    var actionsMenuMobileEnterStandaloneMode: some View {
```
[the remark this answers is not in the document]
Comment (anna1901) [reply AirCasting/SessionViews/SessionHeaderView.swift:143]: hm, from what I see in all of our views the convention is to not make this kind of variables private

File: AirCasting/MobilePeripheralSessionManager.swift [hunk 5]
```diff
@@ -45,29 +46,94 @@ class MobilePeripheralSessionManager {
         }
     }
     
-    func finishActiveSession(centralManger: CBCentralManager) {
-        guard let activePeripheral = activeMobileSession?.peripheral else { return }
-        finishSession(for: activePeripheral, centralManger: centralManger)
+    func finishSession(for peripheral: CBPeripheral, centralManager: CBCentralManager) {
+        if activeMobileSession?.peripheral == peripheral {
```
[the remark this answers is not in the document]
Comment (anna1901) [reply AirCasting/MobilePeripheralSessionManager.swift:50]: I tried that and I think it made the intent less clear so I prefer to leave it as is.

[the remark this answers is not in the document]
Comment (anna1901) [reply AirCasting/MobilePeripheralSessionManager.swift:50]: But in general that'a a good point and I will start to favour this approach :)

File: AirCasting/SessionViews/SessionHeaderView.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (pawelgil) [reply AirCasting/SessionViews/SessionHeaderView.swift:143]: I really don't think this is a convention, just overall `private` negligence in the project, especially views ;) There is no reason to expose such internals to the public. It's not a huge deal with SwiftUI views, but the habit of not marking private stuff will really harm code quality when it comes to business logic classes

**Review by pawelgil — APPROVED** [review]

**Pull request merged**